source/fundamentals/object-encodings.txt

```diff
@@ -0,0 +1,98 @@
+using an optional second parameter of the
+:java-sync-api:`MongoDatabase.getCollection()
+<com/mongodb/client/MongoDatabase.html#getCollection(java.lang.String,java.lang.Class)>`
+method.
+
+To use a class in this way, a :java-docs:`Codec
+<apidocs/bson/org/bson/codecs/Codec.html>` for it must be registered in the
+:java-docs:`CodecRegistry
+<apidocs/bson/org/bson/codecs/configuration/CodecRegistry.html>`
+of the ``MongoCollection``.
+
+Default Codecs
+--------------
+
+By default, every ``MongoCollection`` has ``Codec``s registered for
+three classes:
+
+- ``Document``
+- ``BasicDBObject``
+- ``BsonDocument``
+
+Custom Codecs
+-------------
+
+Applications are free to register ``Codec`` implementations for other
+classes by adding them to the ``CodecRegistry``. You can configure
+``CodecRegistry`` instances at the ``MongoClient``, ``MongoDatabase``,
+and ``MongoCollection`` levels.
+
+To register a ``Codec`` for a ``MongoClient``, use the
+:java-api-sync:`MongoClientSettings
+<com/mongodb/MongoClientSettings.html>` class:
+
+.. code-block:: java
+
+   CodecRegistry codecRegistry =
+      CodecRegistries.fromRegistries(MongoClientSettings.getDefaultCodecRegistry(),
+                fromProviders(PojoCodecProvider.builder().automatic(true).build()));
+   MongoClientSettings settings = MongoClientSettings.builder()
+        .codecRegistry(codecRegistry).build();
+   MongoClient client = MongoClients.create(settings);
+
+To register a ``Codec`` for a ``MongoDatabase``, use the
+:java-sync-api:`withCodecRegistry()
+<com/mongodb/client/MongoDatabase.html#withCodecRegistry(org.bson.codecs.configuration.CodecRegistry)>`
+method:
+
+.. code-block:: java
+
+   CodecRegistry codecRegistry =
+      CodecRegistries.fromRegistries(MongoClientSettings.getDefaultCodecRegistry(),
+                fromProviders(PojoCodecProvider.builder().automatic(true).build()));
+   MongoDatabase database = client.getDatabase("exampleDatabase")
+                               .withCodecRegistry(codecRegistry);
+
+To register a ``Codec`` for a ``MongoCollection``, use the
+:java-sync-api:`withCodecRegistry()
+<com/mongodb/client/MongoCollection.html#withCodecRegistry(org.bson.codecs.configuration.CodecRegistry)>`
+method:
+
+.. code-block:: java
+
+   CodecRegistry codecRegistry =
+      CodecRegistries.fromRegistries(MongoClientSettings.getDefaultCodecRegistry(),
+                fromProviders(PojoCodecProvider.builder().automatic(true).build()));
+   MongoCollection<Document> collection = database.getCollection("exampleCollection")
+                                               .withCodecRegistry(codecRegistry);
+
+UUID Class
+----------
+
+By default, the driver encodes instances of `UUID
+<https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/UUID.html>`__
+using a byte ordering that is not compatible with other MongoDB drivers.
+If your application requires interoperability across multiple drivers,
+you can change the default ``UUID`` encoding with the
```

Comment:
Nitpick: prefer direct imperative language for clarity. E.g.
"... across multiple drivers, use the ``UUID`` encoding provided by the UuidRepresentation.STANDARD Codec as shown in the following example:"